lagom-eventstore-example/lagom-eventstore-impl/src/main/java/com/lightbend/lagom/eventstore/impl/readside/EventStoreRepositoryImpl.java

```diff
@@ -0,0 +1,142 @@
+import com.ibm.event.oltp.EventContext$;
+import com.ibm.event.oltp.EventError;
+import com.ibm.event.oltp.InsertResult;
+import com.lightbend.lagom.eventstore.impl.writeside.HelloEvent;
+import com.typesafe.config.Config;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import scala.Option;
+import scala.collection.IndexedSeq;
+import scala.collection.JavaConversions;
+import scala.collection.Seq;
+import scala.compat.java8.FutureConverters;
+import scala.compat.java8.OptionConverters;
+
+import javax.inject.Singleton;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+@Singleton
+public class EventStoreRepositoryImpl {
+
+
+    private static final String TABLE_NAME = "Greetings";
+    private final EventStoreConfig eventStoreConfig;
+    private ResolvedTableSchema greetingsTable;
+    private EventContext eventCtx;
+
+    public EventStoreRepositoryImpl(com.typesafe.config.Config config) {
+        eventStoreConfig = new EventStoreConfig(config);
+
+    }
+
+    public CompletionStage<Done> ensureSchema() {
+        return CompletableFuture.supplyAsync(() -> {
+
+            ConfigurationReader.setConnectionEndpoints(eventStoreConfig.endpoints);
+
+            String databaseName = eventStoreConfig.databaseName;
+            Option<EventError> maybeErrors = EventContext$.MODULE$.openDatabase(databaseName);
+            // if there was no error opening the DB it means there's a DB
+            if (maybeErrors.isEmpty()) {
+                // drop the DB
+                EventContext.dropDatabase(databaseName);
+            }
+            EventContext.createDatabase(databaseName);
+            return Done.getInstance();
+        });
+    }
+
+    public CompletionStage<Done> ensureTables() {
+        this.eventCtx = EventContext.getEventContext();
+        return CompletableFuture.supplyAsync(() -> {
+            Seq<String> shardingColumns = JavaConversions.asScalaBuffer(Collections.EMPTY_LIST).toSeq();
+            Seq<String> pkColumns = JavaConversions.asScalaBuffer(Arrays.asList("instant", "name")).toSeq();
+
+            StructField[] fields = new StructField[]{
+                    DataTypes.createStructField("name", DataTypes.StringType, false),
+                    DataTypes.createStructField("instant", DataTypes.IntegerType, false)
+            };
+
+            TableSchema greetingsSchema =
+                    new TableSchema(
+                            TABLE_NAME,
+                            new StructType(fields),
+                            shardingColumns,
+                            pkColumns,
+                            Option.<Seq<String>>empty());
+
+            // create the table
+            Optional<EventError> maybeError =
```

Comment:
Any particular reason for converting it to java `Optional`? 
On line https://github.com/lagom/ibm-integration-examples/pull/2/files#diff-a77dd286c1d6fadcb6e10348059ab412R54 Scala `Option` is being used.

Reply:
I tried to expose Scala types as little as possible so you just pointed me to another place I'll have to review. :-D